pkg/oidc/userinfo.go

```diff
@@ -389,7 +389,13 @@ func (u *userinfo) UnmarshalJSON(data []byte) error {
 	if err := json.Unmarshal(data, &a); err != nil {
 		return err
 	}
-	u.Address = a.Address
+
+	if a.Address == nil {
+		u.Address = new(userInfoAddress)
+	} else {
+		u.Address = a.Address
+	}
```

Comment:
what about if we only check for nil here:
```go
if a.Address != nil {
	u.Address = a.Address
}
```
and an additional check in the GetAddress():
```go
func (u *userinfo) GetAddress() UserInfoAddress {
	if u.Address == nil {
		return &userInfoAddress{}
	}
	return u.Address
}
```
this way we can avoid the nil pointer and marshalling the userinfo again would return the same values as we had before unmarshalling

Reply:
I agree. Done